libraries/AP_NavEKF3/AP_NavEKF3_AirDataFusion.cpp

```diff
@@ -444,6 +449,276 @@ void NavEKF3_core::FuseSideslip()
+            const float HK25 = HK9*P[5][23];
+            const float HK26 = HK12*P[5][6] - HK13*P[4][5] + HK13*P[5][22] + HK14*P[2][5] + HK15*P[0][5] + HK16*P[1][5] - HK17*P[3][5] - HK25 + HK9*P[5][5];
+            const float HK27 = HK19*HK9;
+            const float HK28 = HK12*P[6][23] + HK13*P[22][23] - HK13*P[4][23] + HK14*P[2][23] + HK15*P[0][23] + HK16*P[1][23] - HK17*P[3][23] + HK25 - HK9*P[23][23];
+            const float HK29 = HK12*P[2][6] + HK13*P[2][22] - HK13*P[2][4] + HK14*P[2][2] + HK15*P[0][2] + HK16*P[1][2] - HK17*P[2][3] - HK9*P[2][23] + HK9*P[2][5];
+            const float HK30 = HK12*P[1][6] + HK13*P[1][22] - HK13*P[1][4] + HK14*P[1][2] + HK15*P[0][1] + HK16*P[1][1] - HK17*P[1][3] - HK9*P[1][23] + HK9*P[1][5];
+            const float HK31 = HK12*P[3][6] + HK13*P[3][22] - HK13*P[3][4] + HK14*P[2][3] + HK15*P[0][3] + HK16*P[1][3] - HK17*P[3][3] - HK9*P[3][23] + HK9*P[3][5];
+            // const float HK32 = Kaccy/(HK12*HK19*HK20 + HK14*HK19*HK29 + HK15*HK18*HK19 + HK16*HK19*HK30 - HK17*HK19*HK31 + HK22*HK23 - HK23*HK24 + HK26*HK27 - HK27*HK28 + R_ACC);
+
+            innovDragVar.y = (HK12*HK19*HK20 + HK14*HK19*HK29 + HK15*HK18*HK19 + HK16*HK19*HK30 - HK17*HK19*HK31 + HK22*HK23 - HK23*HK24 + HK26*HK27 - HK27*HK28 + R_ACC);
+            if (innovDragVar.y < R_ACC) {
+                // calculation is badly conditioned
+                return;
+            }
+            const float HK32 = Kacc / innovDragVar.y;
+
+            // Observation Jacobians
+            Hfusion[0] = -HK2*HK3;
+            Hfusion[1] = -HK3*HK4;
+            Hfusion[2] = -HK3*HK5;
+            Hfusion[3] = HK3*HK6;
+            Hfusion[4] = HK8;
+            Hfusion[5] = -HK10;
+            Hfusion[6] = -HK11*HK3;
+            Hfusion[22] = -HK8;
+            Hfusion[23] = HK10;
+
+            // Kalman gains
+            // Don't allow modification of any states other than wind velocity at this stage of development - we only need a wind estimate.
+            // See AP_NavEKF3/derivation/generated/acc_bf_generated.cpp for un-implemented Kalman gain equations.
+            Kfusion[22] = -HK22*HK32;
+            Kfusion[23] = -HK28*HK32;
+        }
+
+        innovDrag[axis_index] = predAccel - mea_acc;
+        dragTestRatio[axis_index] = sq(innovDrag[axis_index]) / (25.0f * innovDragVar[axis_index]);
+
+        // if the innovation consistency check fails then don't fuse the sample
+        if (dragTestRatio[axis_index] > 1.0f) {
+            return;
+        }
+
+        // correct the state vector
+        for (uint8_t j= 0; j<=stateIndexLim; j++) {
+            statesArray[j] = statesArray[j] - Kfusion[j] * innovDrag[axis_index];
+        }
+        stateStruct.quat.normalize();
+
+        // correct the covariance P = (I - K*H)*P
+        // take advantage of the empty columns in KH to reduce the
+        // number of operations
+        for (unsigned i = 0; i<=stateIndexLim; i++) {
```

Comment:
we should be using uint8_t here instead of "unsigned"

Reply:
Early EKF development profiling showed the use of unsigned to be faster in this context and this is repeated throughout.

Reply:
ok, although I suspect that uint16_t would be just as fast though.  Anyway not a huge deal..